packages/wonder-blocks-dropdown/components/dropdown.js

```diff
@@ -0,0 +1,234 @@
+// @flow
+
+import * as React from "react";
+import ReactDOM from "react-dom";
+import {StyleSheet} from "aphrodite";
+import type {AriaProps, StyleType} from "@khanacademy/wonder-blocks-core";
+import {getClickableBehavior} from "@khanacademy/wonder-blocks-core";
+import DropdownCore from "./dropdown-core.js";
+import type {Item, DropdownItem} from "../util/types.js";
+import ActionItem from "./action-item.js";
+import OptionItem from "./option-item.js";
+import DropdownAnchor from "./dropdown-anchor.js";
+
+type Props = {|
+    ...AriaProps,
+
+    /**
+     * The child function, that takes in a function to open the dropdown and state and
+     * returns an Element which will become the opener element for the dropdown.
+     */
+    children: (
+        openDropdown: () => void,
+        state: ClickableState,
+    ) => React.Element<any>,
+
+    /**
+     * Can be used to override the state of the Dropdown by parent elemnents
+     */
+    opened?: boolean,
+
+    /**
+     * The items present in the Dropdown
+     */
+    menuItems: Array<Item> | Item,
+
+    /**
+     * A callback that returns items that are newly selected. Use only if this
+     * menu contains select items (and make sure selectedValues is defined).
+     */
+    onChange?: (selectedItems: Array<string>) => mixed,
+
+    /**
+     * The values of the items that are currently selected. Use only if this
+     * menu contains select items (and make sure onChange is defined).
+     */
+    selectedValues?: Array<string>,
+
+    /**
+     * Whether this menu should be left-aligned or right-aligned with the
+     * opener component. Defaults to left-aligned.
+     */
+    alignment?: "left" | "right",
+
+    /**
+     * Test ID used for e2e testing.
+     */
+    testId?: string,
+
+    /**
+     * Optional styling to add to the dropdown wrapper.
+     */
+    dropdownStyle?: StyleType,
+|};
+
+type ClickableState = {|
+    pressed: boolean,
+    hovered: boolean,
+    focused: boolean,
+|};
+
+type State = {|
+    /**
+     * Whether or not the dropdown is currently open.
+     */
+    open: boolean,
+
+    /**
+     * Whether or not last open state change was triggered by a keyboard click.
+     */
+    keyboard?: boolean,
+|};
+
+export default class Dropdown extends React.Component<Props, State> {
```

Comment:
Since a lot of this logic has been lifted from `action-menu.js`, I think a good follow up task would be to refactor `action-menu.js` to use `Dropdown` instead of `DropdownCore`.  Can you create a Jira ticket for this and add a TODO with the ticket number in `action-menu.js`?

Reply:
Friendly ping.